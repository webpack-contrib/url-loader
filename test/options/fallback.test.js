/* eslint-disable
  prefer-destructuring,
*/
import webpack from '@webpack-contrib/test-utils';

describe('Options', () => {
  describe('fallback', () => {
    test('{String}', async () => {
      const config = {
        loader: {
          test: /\.png$/,
          options: {
            limit: 100,
            fallback: 'file-loader',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[0];

      expect(source).toMatchSnapshot();
    });

    // Version 1.0.1 passes options provided to url-loader to the fallback as well, so make sure that still works.
    test('{String}', async () => {
      const config = {
        loader: {
          test: /\.png$/,
          options: {
            limit: 100,
            fallback: 'file-loader',
            name: '[name].[ext]',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[0];

      expect(source).toMatchSnapshot();
    });

    // Test passing explicitly provided options to the fallback loader.
    test('{String}', async () => {
      const config = {
        loader: {
          test: /\.png$/,
          options: {
            limit: 100,
            name: 'name-for-url-loader.[ext]',
            fallback: {
              loader: 'file-loader',
              options: {
                name: 'name-for-file-loader.[ext]',
              },
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[0];

      expect(source).toMatchSnapshot();
    });

    test('{String}', async () => {
      const config = {
        loader: {
          test: /\.png$/,
          options: {
            limit: 100,
            name: 'name-for-url-loader.[ext]',
            fallback: 'file-loader?name=name-for-file-loader.[ext]',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[0];

      expect(source).toMatchSnapshot();
    });
  });
});
