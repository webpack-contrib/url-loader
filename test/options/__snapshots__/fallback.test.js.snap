// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options fallback {String} 1`] = `"module.exports = __webpack_public_path__ + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";"`;

exports[`Options fallback {String} 2`] = `"module.exports = __webpack_public_path__ + \\"file.png\\";"`;

exports[`Options fallback {String} 3`] = `"module.exports = __webpack_public_path__ + \\"name-for-file-loader.png\\";"`;

exports[`Options fallback {String} 4`] = `"module.exports = __webpack_public_path__ + \\"name-for-file-loader.png\\";"`;
